src/main/java/com/devonfw/ide/sonarqube/common/impl/check/DevonNamingConventionClassExtendsClassCheck.java

```diff
@@ -0,0 +1,179 @@
+
+  public abstract void init();
+
+  @Override
+  public void scanFile(JavaFileScannerContext context) {
+
+    this.context = context;
+
+    File file = context.getFile();
+
+    CompilationUnitTree parsedTree = parseJavaFile(file.getPath());
+
+    List<Tree> types = parsedTree.types();
+    ClassTree tree = getTreeInstance(types);
+
+    Kind kind = tree.kind();
+
+    boolean isClass = kind.equals(Tree.Kind.CLASS);
+
+    if (!isClass)
+      return;
+
+    TypeTree superClass = tree.superClass();
+
+    LinkedList<String> superInterfacesNames = new LinkedList<String>();
+    ListTree<TypeTree> superInterfaces = tree.superInterfaces();
+
+    IdentifierTree simpleName = tree.simpleName();
+    String className = simpleName.name();
+
+    this.className = className;
+
+    init();
+
+    for (Iterator<TypeTree> iterator = superInterfaces.iterator(); iterator.hasNext();) {
+      TypeTree typeTree = iterator.next();
+      superInterfacesNames.add(typeTree.toString());
+    }
+
+    String superClassName = null;
+    if (superClass != null) {
+      superClassName = superClass.toString();
+    }
+
+    Symbol.TypeSymbol typeSymbol = tree.symbol();
+
+    if (this.isAbstract != null)
+      if (this.isAbstract) {
+        if (!typeSymbol.isAbstract()) {
+          return;
+        }
+      } else {
+        if (!this.isAbstract) {
+          if (typeSymbol.isAbstract()) {
+            return;
+          }
+        }
+
+      }
+
+    if (superClassName != null) {
+
+      if (superClassName.equals(this.extendedSuperClass)) {
+
+        Pattern pattern = Pattern.compile(this.classSuffixRegEx);
+        Matcher matcher = pattern.matcher(className);
+        boolean endsWith = matcher.find();
+
+        if (!endsWith) {
+          this.context.addIssueOnFile(this, "Classes inheriting from " + this.extendedSuperClass + " should have "
+              + this.classSuffixRegEx + " as prefix");
+        }
+
+        if (this.interfacesToImplement != null) {
+          boolean contains = superInterfacesNames.containsAll(this.interfacesToImplement);
+
+          if (!contains) {
+            this.context.addIssueOnFile(this, "Classes extending " + this.extendedSuperClass
+                + " should implement an interface with the same name except the suffix");
+          }
+        }
+      } else if (Pattern.compile(this.classSuffixRegEx).matcher(superClassName).find()) {
+
+        if (!Pattern.compile(this.classSuffixRegEx).matcher(className).find()) {
+          this.context.addIssueOnFile(this, "If a superclass has" + this.classSuffixRegEx
+              + " as prefix, then the subclass should also have" + this.classSuffixRegEx + " as prefix.");
+
+        }
+      }
+    }
+  }
+
+  private static String getFileContent(File file) throws IOException {
+
+    BufferedReader br = new BufferedReader(new FileReader(file));
+    String output = "";
+    String st;
+    while ((st = br.readLine()) != null)
+      output += st + "\n";
```

Comment:
is that used at all?
If yes, there are more efficient ways in Java to read a file into a String...

Reply:
You're right. You're right. That is no longer needed. The parser method can also handle files.